Add dependency on grpc-core to the main README.md

The "Download" section starts with a Maven pom file excerpt containing some dependencies. Note that grpc-netty-shaded doesn't include some classes that are available in grpc-core, for example, `io.grpc.internal.AbstractServerImplBuilder`. I suggest to add dependency on grpc-core to avoid confusion. 
(I've spent quite some time trying to figure out why example-tls builds with Maven but my project doesn't... Turned out the difference is that my project uses netty-shaded.)

During compilation.
Here's how you can reproduce the problem:
- Take `example-tls` with Maven (it depends on `grpc-netty`, not `grpc-netty-shaded`).
- Change `pom.xml` to depend on `grpc-netty-shaded`.
- Fix the packages to point to shaded packages ( `s/import io.grpc.netty/import io.grpc.netty.shaded.io.grpc.netty`/`, `s/import io.netty/import io.grpc.netty.shaded.io.netty/`).
- Compile.

Observe:
```
[ERROR] X:\src\grpc-java\examples\example-tls\src\main\java\io\grpc\examples\helloworldtls\HelloWorldServerTls.java:[69,35] error: cannot access AbstractServerImplBuilder
[ERROR]   class file for io.grpc.internal.AbstractServerImplBuilder not found
X:\src\grpc-java\examples\example-tls\src\main\java\io\grpc\examples\helloworldtls\HelloWorldClientTls.java:[63,32] error: cannot access AbstractManagedChannelImplBuilder
```

--- a/netty/shaded/build.gradle
+++ b/netty/shaded/build.gradle
@@ -56,7 +56,7 @@ publishing {
                     dependencyNode.appendNode('artifactId', dep.name)
                     def version = (dep.name == 'grpc-core') ? '[' + dep.version + ']' : dep.version
                     dependencyNode.appendNode('version', version)
-                    dependencyNode.appendNode('scope', 'runtime')
+                    dependencyNode.appendNode('scope', 'compile')
                 }
             }
         }

